lib/widgets/message_list.dart

```diff
@@ -186,7 +186,7 @@ class _MessageListState extends State<MessageList> with PerAccountStoreAwareStat
   }
 
   @override
-  void onNewStore() {
+  void onNewStore() { // TODO(#464) try to keep using old model until new one gets messages
     model?.dispose();
```

Comment:
Hmm. `model`, here, refers to the message-list view on the old per-account store. In the case of refreshing a dead event queue, those things (the old message-list view and the old per-account store) will have already been disposed, by the `UpdateMachine`, and so I think it's not appropriate to do anything with them, including this `dispose` call.

I was alerted to this by the following `assert` in `PerAccountStore.unregisterMessageList`, which was throwing in my manual testing:

```dart
  void unregisterMessageList(MessageListView view) {
    final removed = _messageListViews.remove(view);
    assert(removed);
  }
```

One maybe-naïve fix is to just remove the `model?.dispose();` line in this `onNewStore`. But the `TODO(#464)` seems fundamentally impossible to accomplish as long as the old model is disposed by the `UpdateMachine` before `_MessageListState` even learns (by a call to its `onNewStore`) that it has to refetch messages.

Reply:
> I was alerted to this by the following `assert` in `PerAccountStore.unregisterMessageList`, which was throwing in my manual testing:

Hmm, good catch, thanks.


> will have already been disposed, by the `UpdateMachine`, and so I think it's not appropriate to do anything with them, including this `dispose` call.

In general `dispose` should be idempotent — the usual spec for it is "it's no longer valid to do anything with this object, except call `dispose` again".

… Or maybe not. Looking around for examples, both `ChangeNotifier.dispose` and `State.dispose` have asserts that will fail if you dispose twice.


> One maybe-naïve fix is to just remove the `model?.dispose();` line in this `onNewStore`. But the `TODO(#464)` seems fundamentally impossible to accomplish as long as the old model is disposed by the `UpdateMachine` before `_MessageListState` even learns (by a call to its `onNewStore`) that it has to refetch messages.

Yeah, #464 will definitely call for the `PerAccountStore.dispose` to get delayed. One thing that'll be a bit tricky in implementing that issue will be to ensure that it does eventually get disposed — something like, when it's already stale and then its last `MessageListView` goes away.

But in the meantime, this code can be written for the way that `PerAccountStore.dispose` currently works — so it can just drop this `dispose` call.